lib/payment_processor/braintree/one_click_from_uri.rb

```diff
@@ -48,7 +48,14 @@ def options
     end
 
     def payment_method_id
-      member.customer.payment_methods.stored.first.try(:id) if member.try(:customer)
+      cookied_payment_methods.split(',').each do |token|
+        pm = member.customer.payment_methods.find_by(token: token, cancelled_at: nil, store_in_vault: true)
+        if pm && pm.expiration_date.to_date > Date.today
```

Comment:
Not a big deal, but you could check for the expiration date on the SQL query too.